components/feature/qr/src/test/java/mozilla/components/feature/qr/QrFragmentTest.kt

```diff
@@ -617,4 +619,68 @@ class QrFragmentTest {
         assertSame(bitmap, imageCaptor.allValues[0])
         assertSame(invertedBitmap, imageCaptor.allValues[1])
     }
+
+    @Test
+    @Suppress("DEPRECATION")
+    fun `getScreenRotation returns correct display rotation`() {
+        val mockActivity: FragmentActivity = mock()
+        val mockManager: WindowManager = mock()
+        val mockDisplay: Display = mock()
+
+        val testRotation = Surface.ROTATION_90
+
+        whenever(mockActivity.windowManager).thenReturn(mockManager)
+        whenever(mockManager.defaultDisplay).thenReturn(mockDisplay)
+        whenever(mockDisplay.rotation).thenReturn(testRotation)
+
+        val listener = mock<QrFragment.OnScanCompleteListener>()
+        val qrFragment = spy(QrFragment.newInstance(listener))
+        whenever(qrFragment.activity).thenReturn(mockActivity)
+
+        val rotation = qrFragment.getScreenRotation()
+
+        assertEquals(testRotation, rotation)
+    }
+
+    @Test
+    @Suppress("DEPRECATION")
+    fun `configureTransform uses getScreenRotation method to get rotation`() {
```

Comment:
See https://github.com/mozilla-mobile/android-components/pull/9759#discussion_r588687156

Reply:
This is not actually the same as  https://github.com/mozilla-mobile/android-components/pull/9759#discussion_r588687156, because we just check if the new method is used when calculating transform. No idea how I would rename this.

Reply:
Thanks for the clarification, yeah I'm not sure how to name it neither, I guess It could that we shouldn't be checking for functions call but rather expected behavior, but I'm aware that checking for behavior is not always an option. Maybe @csadilek could have some ideas. Otherwise we could keep them.

Reply:
Agreed that the QR functionality is pretty hard to unit test but this test and the two below really don't seem to add much protection. They just verify that `getScreenRotation` is called unconditionally. I would probably not even add those. To get something out of those tests we'd need to make assertions on the matrix passed to `setTransform`.

Reply:
I just added those tests to ensure future changes will use the same methods that handle deprecations. I can remove them if they do not add value. Also, I would open a tech-debt new issue for extending coverage on QR Fragment code.